pkg/api/const.go

```diff
@@ -35,6 +35,7 @@ const (
 	AKSUbuntu1604     Distro = "aks-ubuntu-16.04"
 	AKSUbuntu1804     Distro = "aks-ubuntu-18.04"
 	ACC1604           Distro = "acc-16.04"
+	WindowsDistro     Distro = "windows"
```

Comment:
Not sure re-using the `Distro` property makes sense here, since "distro" is [commonly known to mean Linux distribution.](https://en.wikipedia.org/wiki/Linux_distribution).

Reply:
I think I'd be ok overloading here here.
Lets be more specific so we can support multiple releases of windows server.
How about AKSWindowsServer2019 here to match the

Reply:
That doesn't feel great to me... any way we could rename the property to something more generic or create a separate string type for Windows? @jackfrancis thoughts?

If we do decide to stick with this we also need to change the comment above the "Distro" type - right now it's `Distro represents Linux distro to use for Linux VMs`.

Reply:
I will update the comment for the `Distro` type and rename `WindowsDistro` to `AKSWindowsServer2019`.
I also will add `Distro` in `WindowsProfile` so users can specify it.

Reply:
I'm still not sure we need any changes to get the desired behavior.

If you specify the following in the cluster definition you'll get the default up-to-date values.
"windowsProfile": {
      "adminUsername": "azureuser",
      "adminPassword": "####",
      "sshEnabled": true,
      "enableAutomaticUpdates": false
    }
I believe AKS was using hard-coded values because at some point in early 2019 there was some stability issues with windows containers and AKS wanted to hard-code the image versions instead of specifying "latest"

Today we manage the defaults in aks-engine and before updating the defaults to point to new image versions.

Are there cases where this won't work? I just want to understand the use cases here.

@PatrickLang what do you think here?

Reply:
After checking the code, I introduce one new variable `WindowsOSVersion` for Windows OS images because `DistroValues` are defined in defaults.go and used in many places for Linux.